### ISSUE TITLE
Resources and slot times

### ISSUE THREAD

**Issue by M0in4** [werner-scholtz/kalender#30]
Is it possible to get the appointment for X resources where the appointment are divided inside the day view for each resource on top and is It possible to add minutes to timeline to view hours with interval? like 15 minutes

**Comment by M0in4** [comment]
I've already tried to add minutes but the problem is that we have to edit not only the Timeline but also the line that are drawn near each hour because I need the line each 15 minutes like.

For the resources the example is shown below:


![300x0w](https://github.com/werner-scholtz/kalender/assets/52939366/d7ed7b66-3591-4667-af39-80d8e690c5c9)

**Comment by werner-scholtz** [comment]
Aah okay to add the lines you will need a custom hourlineBuilder as well, here is the [default](https://github.com/werner-scholtz/kalender/blob/c4c77d703c9f2744f4eed8006d880abde87657a0/lib/src/components/general/hour_line/hour_lines.dart#L6).

**Comment by M0in4** [comment]
Ok and regarding the resources?
As you can see at the top of calendar we got two names "Massimiliano" and "Giada" where appointments are divided by resources too

**Comment by werner-scholtz** [comment]
As for the "resources" it is a bit more complicated but possible.
For the resource headers I suggest you add that to the bottom of the CalendarHeader, is this acceptable ?
To layout the tiles underneath the resource header you will need to create and pass in a custom EventGroupLayoutDelegate that can layout the tiles in the correct x positions. Some examples [EventGroupOverlapLayoutDelegate](https://github.com/werner-scholtz/kalender/blob/c4c77d703c9f2744f4eed8006d880abde87657a0/lib/src/components/layout_delegates/event_group_layout.dart#L98) , [EventGroupBasicLayoutDelegate](https://github.com/werner-scholtz/kalender/blob/c4c77d703c9f2744f4eed8006d880abde87657a0/lib/src/components/layout_delegates/event_group_layout.dart#L166)

**Comment by x-ji** [comment]
Following this issue since I'd also need something similar for my app. This library seems very feature-complete and promising, thanks for authoring it. The multi-resource-in-a-single-day arrangement would be the last piece I need : )

Just to clarify @werner-scholtz , in this comment https://github.com/werner-scholtz/kalender/issues/30#issuecomment-1957035587, do you mean that it should already be possible to implement such a view in my own app using the current version of the package, where I create my own EventGroupLayoutDelegate by extending the EventGroupLayoutDelegate class? i.e. everybody doesn't have to wait for this to become an official layout type before they try to implement it in their own app, right?

**Comment by werner-scholtz** [comment]
Hi @x-ji yes it should be possible to do this with the current package.

On a side note I'm currently reworking the package and once released it will most likely contain breaking changes related to this issue.

**Comment by x-ji** [comment]
@werner-scholtz Sounds good. Thanks for the heads-up. 

Would you know how long it will be before the breaking version is released, and how much work it's likely to be to adapt the logic? I'd like to decide whether to write it now or whether to focus on other parts of the app for now and wait for the release : ) Thanks!

**Comment by Ashotjlavyan** [comment]
@werner-scholtz 

How can we handle dates in the resource calendar to allow for repeating dates? As shown in the image, the same date repeats for different resources. For example, we have two Mondays, two Tuesdays, etc., for two resources. If there are three resources, the dates will be repeated three times. How can we implement this?

**Comment by werner-scholtz** [comment]
@Ashotjlavyan @M0in4 @x-ji 

Once the new version is released, I will add an example of how to do something like this.

**Comment by x-ji** [comment]
@werner-scholtz Just to confirm, has the breaking change you mentioned [here](https://github.com/werner-scholtz/kalender/issues/30#issuecomment-2200625323) happened? I see there have been constant updates to the package. Thanks!

**Comment by x-ji** [comment]
(I see the API of `CalendarView` is already different (e.g. there's no `layoutDelegates` anymore), so I assume it should be safe to assume the new API will be the basis going forward and I can try to implement this custom layout on top of that in the latest version?

**Comment by ashot3** [comment]
@werner-scholtz Do you have any example of how we can do this? The solution doesn’t seem straightforward

**Comment by werner-scholtz** [comment]
Hi, I have not gotten around to do an example of this yet sorry.

**Comment by x-ji** [comment]
A separate question: I tried to manually change the index to `person: people[1]` and tried to create events, but it errored. Apparently the tilewidth became a negative number from this logic in `examples/advanced_example/lib/layout_strategy.dart`:

```
        // Calculate the width of the tile.
        var tileWidth = childWidth;
        if (overlapsRight.isEmpty) {
          tileWidth = rect.width - tileXOffset;
        }

        // Layout the tile.
        layoutChild(id, BoxConstraints.tightFor(width: tileWidth, height: data.height));
```

The `tileXOffset` is much larger than `rect.width`, resulting in `tileWidth` becoming around -748.

**Comment by werner-scholtz** [comment]
Yeah I need to make some changes to the callbacks, so it is easier to add events to the correct person. https://github.com/werner-scholtz/kalender/pull/219

**Comment by x-ji** [comment]
Great. Thanks again for adding support for this 🙌 , which I'd imagine will be a relatively frequent use case : ) I'll keep an eye on the updates.

**Comment by x-ji** [comment]
Thanks @werner-scholtz . It worked really well!
